### DISCUSSION TITLE
YNL test fails

### DISCUSSION THREAD

**Discussion by MohanPrasad-J** [linux-netdev/nipa#37] [General]
While I try to run ingest_mdir.py for a series of patches. The YNL test always fails throwing up an error log(stderr) with the below warning.
**netdev.c: In function ‘main’:
netdev.c:82:9: warning: ignoring return value of ‘scanf’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
   82 |         scanf("%d", &ifindex);
      |         ^~~~~~~~~~~~~~~~~~~~~**

netdev.c being in the directory /tools/net/ynl/samples/netdev.c
I observed that these tests are passing in the patchwork system.
Since this is not a part of change I did, The YNL test should not fail for my series of patches.
How to resolve this?

**Comment by kuba-moo** [comment]
I submitted a fix: https://lore.kernel.org/20240828173609.2951335-1-kuba@kernel.org
Just to be clear, are you trying to run the tests to test your own submissions? Or for compliance with the upstream testing requirement for drivers?

**Reply by MohanPrasad-J** [reply]
I am running the tests for compliance with the upstream testing requirement.

**Reply by kuba-moo** [reply]
I was afraid you'd say that :) Please read the announcement carefully https://lore.kernel.org/all/20240425114200.3effe773@kernel.org/
You don't have to run the basic build / code checker tests. As **point 2** in the announcement states, you have to run `drivers/net` and `drivers/net/hw` against your device.
